File: tests/src/main/java/com/github/klee0kai/test/gc/di/GComponent.java
package com.github.klee0kai.test.gc.di;

import com.github.klee0kai.stone.annotations.component.*;
import com.github.klee0kai.stone.interfaces.IComponent;
import com.github.klee0kai.test.cache.di.CacheDataModule;
import com.github.klee0kai.test.gc.di.ann.AppGcScope;
import com.github.klee0kai.test.gc.di.ann.ContextGcScope;


@Component
public interface GComponent extends IComponent {

    AppModule app();

    CacheDataModule data();


    @GcAllScope
    void gcAll();

    @GcWeakScope
    void gcWeak();

    @GcSoftScope
    void gcSoft();

    @GcStrongScope
    void gcStrong();

    @AppGcScope
    void gcApp();

    @ContextGcScope
    void gcContext();

    @AppGcScope
    @ContextGcScope
    void gcAppAndContext();
}

File: test_kotlin/src/main/kotlin/com/github/klee0kai/test/gc/di/GComponent.kt
package com.github.klee0kai.test.gc.di

import com.github.klee0kai.stone.annotations.component.*
import com.github.klee0kai.stone.interfaces.IComponent
import com.github.klee0kai.test.cache.di.CacheDataModule
import com.github.klee0kai.test.gc.di.ann.AppGcScope
import com.github.klee0kai.test.gc.di.ann.ContextGcScope

@Component
interface GComponent : IComponent {
    fun app(): AppModule
    fun data(): CacheDataModule

    @GcAllScope
    fun gcAll()

    @GcAllScope
    @SwitchCache(cache = SwitchCache.CacheType.Weak)
    fun allWeak()

    @GcWeakScope
    fun gcWeak()

    @GcSoftScope
    fun gcSoft()

    @GcStrongScope
    fun gcStrong()

    @AppGcScope
    fun gcApp()

    @ContextGcScope
    fun gcContext()

    @AppGcScope
    @ContextGcScope
    fun gcAppAndContext()
}